cli/src/main/java/com/devonfw/tools/ide/tool/mvn/Mvn.java

```diff
@@ -154,17 +152,21 @@ private String getEncryptedPassword(String variable) {
 
     String input = this.context.askForInput("Please enter secret value for variable " + variable + ":");
 
-    ProcessContext pc = this.context.newProcess().executable("mvn");
-    pc.addArgs("--encrypt-password", input);
-    pc.addArg(getSettingsSecurityProperty());
-    ProcessResult result = pc.run(ProcessMode.DEFAULT_CAPTURE);
-
-    String encryptedPassword = result.getOut().get(0);
+    String encryptedPassword = retrievePassword("--encrypt-password", input);
     this.context.info("Encrypted as " + encryptedPassword);
 
     return encryptedPassword;
   }
 
+  private String retrievePassword(String args, String input) {
+
+    ProcessResult result = runTool(ProcessMode.DEFAULT_CAPTURE, ProcessErrorHandling.LOG_WARNING, this.context.newProcess(), args, input,
+        getSettingsSecurityProperty());
+
+    IdeSubLogger logger = this.context.level(IdeLogLevel.WARNING);
+    return result.getSingleOutput(logger);
```

Comment:
IMHO we should provide `null` instead of `logger` here.
The old code was throwing an `IndexOutOfBoundsException` in the case that no output was present and was simply ignoring if we have more output than expected.

(After thinking about this, I also wonder why we pass an `IdeSubLogger` as an argument and not just an `IdeLogLevel`.
Maybe my usecase what to privde a mocked `IdeSubLogger` for testing... That make sense...)

Reply:
Adjusted.